component/module/git/git.go

```diff
@@ -91,8 +92,16 @@ func New(o component.Options, args Arguments) (*Component, error) {
 		argsChanged: make(chan struct{}, 1),
 	}
 
+	// Only acknowledge the error from Update if it's not a
+	// vcs.UpdateFailedError; vcs.UpdateFailedError means that the Git repo
+	// exists but we were just unable to update it.
 	if err := c.Update(args); err != nil {
-		return nil, err
+		if errors.As(err, &vcs.UpdateFailedError{}) {
+			level.Error(c.log).Log("msg", "failed to update repository", "err", err)
+			c.updateHealth(err)
```

Comment:
This shouldn't be necessary since the call to `Update` will already update the health for you if it returns an error.

Reply:
Oh, I wasn't aware of this internal detail :D

Removed `updateHealth` in latest commit